core/cmd/app_test.go

```diff
@@ -136,32 +237,134 @@ func Test_initServerConfig(t *testing.T) {
 		{
 			name: "failed to read secrets",
 			args: args{
-				opts:        new(chainlink.GeneralConfigOpts),
-				fileNames:   []string{makeTestFile(t, testConfigFileContents, "test.toml")},
-				secretsFile: "/doesnt-exist",
+				opts:         new(chainlink.GeneralConfigOpts),
+				fileNames:    []string{testtomlutils.WriteTOMLFile(t, testConfigFileContents, "test.toml")},
+				secretsFiles: []string{"/doesnt-exist"},
 			},
 			wantErr: true,
 		},
 		{
 			name: "reading secrets",
 			args: args{
-				opts:        new(chainlink.GeneralConfigOpts),
-				fileNames:   []string{makeTestFile(t, testConfigFileContents, "test.toml")},
-				secretsFile: makeTestFile(t, testSecretsFileContents, "test_secrets.toml"),
+				opts:         new(chainlink.GeneralConfigOpts),
+				fileNames:    []string{testtomlutils.WriteTOMLFile(t, testConfigFileContents, "test.toml")},
+				secretsFiles: []string{testtomlutils.WriteTOMLFile(t, testSecretsFileContents, "test_secrets.toml")},
 			},
 			wantCfg: withDefaults(t, testConfigFileContents, testSecretsRedactedContents),
 		},
+		{
+			name: "reading multiple secrets",
+			args: args{
+				opts:      new(chainlink.GeneralConfigOpts),
+				fileNames: []string{testtomlutils.WriteTOMLFile(t, testConfigFileContents, "test.toml")},
+				secretsFiles: []string{
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Database: testSecretsFileContentsComplete.Database}}, "test_secrets1.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Explorer: testSecretsFileContentsComplete.Explorer}}, "test_secrets2.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Password: testSecretsFileContentsComplete.Password}}, "test_secrets3.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Pyroscope: testSecretsFileContentsComplete.Pyroscope}}, "test_secrets4.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Prometheus: testSecretsFileContentsComplete.Prometheus}}, "test_secrets5.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Mercury: testSecretsFileContentsComplete.Mercury}}, "test_secrets6.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Mercury: additionalMercurySecrets}}, "test_secrets6a.toml"),
+					testtomlutils.WriteTOMLFile(t, chainlink.Secrets{Secrets: toml.Secrets{Threshold: testSecretsFileContentsComplete.Threshold}}, "test_secrets7.toml"),
+				},
+			},
+			wantCfg: withDefaults(t, testConfigFileContents, testSecretsRedactedContentsComplete),
```

Comment:
Maybe I lost the thread somewhere:
1. Why do we still need to marshal redacted values? I thought that reading real values from files would work around that.
2. If we are stuck with this, then can we at least use the redacted Go values for both parts (both `testSecretsRedactedContentsComplete` instead of `testSecretsFileContentsComplete` too)? I think it will be confusing for future readers to see real values being serialized  and implicitly converted.

Reply:
I was sticking with the earlier pattern, but I agree that reading real values from files is a better approach. I've updated in the next revision, thanks!